File: Miscelaneous/conta_bancaria.py
class ContaBancaria:
    def __init__(self, titular):
        self.__titular = titular
        self.__saldo = 0.0

    def __validar_valor(self, valor):
        if valor <= 0:
            raise ValueError("O valor deve ser maior que zero.")

    def depositar(self, valor):
        self.__validar_valor(valor)
        self.__saldo += valor

    def sacar(self, valor):
        self.__validar_valor(valor)
        if valor > self.__saldo:
            return False
        self.__saldo -= valor
        return True

    def consultar_saldo(self):
        return self.__saldo

    @property
    def titular(self):
        return self.__titular

File: Miscelaneous/ContaBancaria.java
public class ContaBancaria {
    private final String titular;
    private double saldo;

    public ContaBancaria(String titular) { this.titular = titular; }

    public void depositar(double valor) {
        validarValor(valor);
        saldo += valor;
    }

    public boolean sacar(double valor) {
        validarValor(valor);
        if (valor > saldo) return false;
        saldo -= valor;
        return true;
    }

    public double consultarSaldo() { return saldo; }
    public String getTitular() { return titular; }

    private void validarValor(double valor) {
        if (valor <= 0) throw new IllegalArgumentException("O valor deve ser maior que zero.");
    }
}